File: Data Structures & Algorithms/lowest-common-ancestor-of-a-binary-tree-iii/submission-0.java
/*
// Definition for a Node.
class Node {
    public int val;
    public Node left;
    public Node right;
    public Node parent;
};
*/

class Solution {
    public Node lowestCommonAncestor(Node p, Node q) {
        Set<Node> pop = new HashSet<>();

        while(p != null){
            pop.add(p);
            p = p.parent;
        }

        while(q != null){
            if (pop.contains(q)){
                return q;
            }
            q = q.parent;
        }
return null;    
}
}

File: Data Structures & Algorithms/reverse-string/submission-0.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        if (s.size() == 0 || s.size() == 1){
            return ;
        }
        int i = 0;
        int j = s.size() - 1 ;

        while(j > i){
            auto t = s[i];
            s[i] = s[j];
            s[j] = t;
            j--;
            i++;
        }
    }
};